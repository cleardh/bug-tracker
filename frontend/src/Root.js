import React from 'react';
import { Provider } from 'react-redux';
import { configureStore, combineReducers } from '@reduxjs/toolkit';

// Reducers
import authReducer from './Controllers/Redux/authSlice';
import bugReducer from './Controllers/Redux/bugSlice';
import userReducer from './Controllers/Redux/userSlice';

export default ({ children }) => {
    // Redux configuration
    const reducer = combineReducers({
        auth: authReducer,
        bugs: bugReducer,
        user: userReducer
    });
    
    const store = configureStore({
        reducer
    });

    return (
        <Provider store={store}>
            {children}
        </Provider>
    );
}