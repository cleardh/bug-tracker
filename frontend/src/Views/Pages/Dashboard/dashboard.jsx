import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useHistory } from 'react-router-dom';
import { getBugs } from '../../../Controllers/Redux/bugSlice';
import Card from '../../Components/Dashboard/card';

export default () => {
    const history = useHistory();
    const dispatch = useDispatch();
    const bugs = useSelector(state => state.bugs);
    let highCount = null;
    let midCount = null;
    let lowCount = null;

    const filterBugs = (priority) => {
        return bugs.filter(bug => bug.priority === priority);
    }

    if (bugs !== undefined) {
        highCount = filterBugs(1);
        midCount = filterBugs(2);
        lowCount = filterBugs(3);
    }

    useEffect(() => {
        dispatch(getBugs());
    }, [bugs === undefined]);

    const redirect = (priority) => {
        history.push({
            pathname: '/viewbugs',
            state: { priority }
        });
    }

    Card.displayName = 'Card';

    return (
        <div className='page-container'>
            <Card priority='1' count={highCount.length} clicked={() => redirect(1)} />
            <Card priority='2' count={midCount.length} clicked={() => redirect(2)} />
            <Card priority='3' count={lowCount.length} clicked={() => redirect(3)} />
        </div>
    )
}
