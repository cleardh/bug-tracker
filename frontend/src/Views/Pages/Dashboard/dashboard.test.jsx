import Root from '../../../Root';
import Dashboard from './dashboard';

it('renders bug cards', () => {
    const wrapper = shallow(
        <Root>
            <Dashboard />
        </Root>
    );
    expect(wrapper).toMatchSnapshot();
});

it('formats bug card components', () => {
    const wrapper = mount(
        <Root>
            <Dashboard />
        </Root>
    );
    const totalText = wrapper.find('h2').length;
    const countText = wrapper.find('p').length;
    expect(totalText).toEqual(3);
    expect(countText).toEqual(3);
});

// it('redirects to viewBugs', () => {
//     const wrapper = mount(
//         <Root>
//             <Dashboard />
//         </Root>
//     );
//     wrapper.find('Card').first().simulate('click');
//     wrapper.update();
// });