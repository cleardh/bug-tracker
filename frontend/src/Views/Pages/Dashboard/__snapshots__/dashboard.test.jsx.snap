// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders bug cards 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <_default />
</Provider>
`;
